frontend/src/pages/notebookController/screens/admin/StopAllServersButton.tsx

```diff
@@ -1,54 +1,34 @@
 import * as React from 'react';
 import { Button } from '@patternfly/react-core';
 import { User } from './types';
-import { deleteNotebook } from '../../../../services/notebookService';
-import useNamespaces from '../../useNamespaces';
-import { allSettledPromises } from '../../../../utilities/allSettledPromises';
-import { Notebook } from '../../../../types';
+import { NotebookAdminContext } from './NotebookAdminContext';
+import { Notebook } from 'types';
 
 type StopAllServersButtonProps = {
   users: User[];
 };
 
-const stopServers = async (
-  projectName: string,
-  serverList: User['serverStatus'][],
-): Promise<void> => {
-  return allSettledPromises<Notebook | void>(
-    serverList.map((serverStatus) => {
-      const notebookName = serverStatus.notebook?.metadata.name || '';
-      if (!notebookName) return Promise.resolve();
-      return deleteNotebook(projectName, notebookName);
-    }),
-  ).then(() => {
-    serverList.forEach((serverStatus) => {
-      serverStatus.forceRefresh();
-    });
-  });
-};
-
 const StopAllServersButton: React.FC<StopAllServersButtonProps> = ({ users }) => {
-  const { notebookNamespace } = useNamespaces();
-  const activeNotebooks = users
+  const activeServers = users
     .filter((user) => user.serverStatus.notebook)
     .map((user) => user.serverStatus);
-  const serverCount = activeNotebooks.length;
-  const [isDeleting, setIsDeleting] = React.useState(false);
+  const activeNotebooks = users.reduce((prev, curr) => {
+    if (curr.serverStatus.notebook) {
+      prev.push(curr.serverStatus.notebook);
+    }
+    return prev;
+  }, [] as Notebook[]);
+  const serverCount = activeServers.length;
+  const { setServerStatuses, setNotebooksToStop } = React.useContext(NotebookAdminContext);
 
   return (
     <Button
       variant="secondary"
       isDanger
-      isDisabled={serverCount === 0 || isDeleting}
+      isDisabled={serverCount === 0}
       onClick={() => {
-        setIsDeleting(true);
-        stopServers(notebookNamespace, activeNotebooks)
-          .then(() => {
-            setIsDeleting(false);
-          })
-          .catch(() => {
-            setIsDeleting(false);
-          });
+        setNotebooksToStop(activeNotebooks);
+        setServerStatuses(activeServers);
```

Comment:
Same here, you already had the `activeServers` -- there is no need to filter down to notebooks